sys/src/libmach/6.c

```diff
@@ -10,10 +10,10 @@
 /*
  * amd64 definition
  */
-#include <u.h>
+#include <amd64/include/u.h>
```

Comment:
I think you're missing the point. libmach is compiled for all architectures. So if you're going to compile 6.c for the riscv, then you need to explicitly include amd64/include/u.h in 6c to get the right register set. The -I /$ARCH is already set up and working, and if you just use u.h for 6.c, you get the riscv register set definitions instead of the ones you want for 6.c. 

For each architecture type in libmach, e.g. 6.c, you have to specify the full path of the u.h, since the u.h might be for a different ARCH than the one you're on. Hence this required change.